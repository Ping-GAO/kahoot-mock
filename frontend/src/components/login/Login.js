import React, { useState } from "react";
import { Button, Form, FormGroup, Label, Input } from "reactstrap";
import { API_URL } from "../../constants";
import "./Login.css";
const Login = () => {
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();
  const handleSubmit = (event) => {
    event.preventDefault();
    console.log({ email: email, password: password });
    fetch(`${API_URL}/admin/auth/login`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ email: email, password: password }),
    })
      .then((res) => res.json())
      .then((data) => console.log(data));
  };

  return (
    <Form className="form-login" onSubmit={handleSubmit}>
      <FormGroup>
        <Label for="exampleEmail">Email</Label>
        <Input
          type="email"
          name="email"
          placeholder="with a placeholder"
          onChange={(event) => setEmail(event.target.value)}
        />
      </FormGroup>
      <FormGroup>
        <Label for="examplePassword">Password</Label>
        <Input
          type="password"
          name="password"
          placeholder="password placeholder"
          onChange={(event) => setPassword(event.target.value)}
        />
      </FormGroup>
      <Button>Submit</Button>
    </Form>
  );
};

export default Login;
