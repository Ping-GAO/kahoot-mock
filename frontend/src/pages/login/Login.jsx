// import React, { useState } from "react";
import React from "react";
// import "./Login.css";
// import { useDispatch } from "react-redux";
// import { useHistory } from "react-router-dom";
// import { login } from "../../redux/actions";

const Login = () => {
    // const [email, setEmail] = useState();
    // const [password, setPassword] = useState();
    // const dispatch = useDispatch();
    // const history = useHistory();
    return (
    // <Form
    //   className="form-login"
    //   onSubmit={e => {
    //     e.preventDefault();
    //     dispatch(login(email, password));
    //     history.push("/home");
    //   }}
    // >
    //   <FormGroup>
    //     <Label for="exampleEmail">Email</Label>
    //     <Input
    //       type="email"
    //       name="email"
    //       placeholder="with a placeholder"
    //       onChange={event => setEmail(event.target.value)}
    //     />
    //   </FormGroup>
    //   <FormGroup>
    //     <Label for="examplePassword">Password</Label>
    //     <Input
    //       type="password"
    //       name="password"
    //       placeholder="password placeholder"
    //       onChange={event => setPassword(event.target.value)}
    //     />
    //   </FormGroup>
    //   <Button type="submit">Submit</Button>
    // </Form>
        <div>fuck</div>
    );
};

export default Login;
