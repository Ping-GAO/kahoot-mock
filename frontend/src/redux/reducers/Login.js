import { userConstants } from "../actionTypes";

// check the localStorage see if user already logged in
let accessToken = JSON.parse(localStorage.getItem("accessToken"));

// Shorthand property names stynex
const initialState = accessToken
  ? { loggedIn: true, accessToken }
  : { loggedIn: false };

// given the current state and action return the next state
export const authentication = (state = initialState, action) => {
  switch (action.type) {
    case userConstants.LOGIN_REQUEST:
      return {
        loggingIn: true,
        user: action.user,
      };
    case userConstants.LOGIN_SUCCESS:
      return {
        loggedIn: true,
        user: action.user,
      };
    case userConstants.LOGIN_FAILURE:
      return {};
    case userConstants.LOGOUT:
      return {};
    default:
      return state;
  }
};
