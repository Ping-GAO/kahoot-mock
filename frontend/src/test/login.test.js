import {mount} from 'enzyme';
import React from "react";
import {Provider} from "react-redux";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import store from "../redux/stores";
import Login from "../pages/login/Login";


// test login page
describe('Login Page', () => {


    let loginPage;
    let wrapper;
    beforeEach(()=>{
        wrapper= mount(
            <Provider store={store}>
                <Login/>
            </Provider>);
        loginPage = wrapper.children();
    });
    afterEach(() => {
        wrapper.unmount();
	 });

    
    it('render a button to login', () => {
       
       
        const loginButton = loginPage.find(Button);
        expect(loginButton.text()).toBe("Log in");
        expect(loginButton).toMatchSnapshot();
    })

    it('render a textfield to input user email', () => {
      
        expect(loginPage.find(TextField).first().text()).toBe("Email *");
    });

    it('render a textfield to input user password ', () => {
      
        expect(loginPage.find(TextField).last().text()).toBe("Password *");
    });
    
    

    describe('my sweet test', () => {
        it('clicks it', () => {
		  
		   
		   console.log();
            //    const spy = jest.spyOn(instance, 'myClickFunc')
	   
            //    instance.forceUpdate();    
	   
            //    const p = app.find('.App-intro')
            //    p.simulate('click')
            //    expect(spy).toHaveBeenCalled()
        })
	   })
	

        
        
})