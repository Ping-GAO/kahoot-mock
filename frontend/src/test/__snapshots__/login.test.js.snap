// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page render a button to login 1`] = `ReactWrapper {}`;
